features/mbedtls/platform/inc/platform_alt.h

```diff
@@ -20,50 +20,48 @@
 
 #ifndef __PLATFORM_ALT__
 #define __PLATFORM_ALT__
-#include "cc_platform.h"
-#include "crys_rnd.h"
-
+#include "platform_mbed.h"
+#if defined(MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT)
+#include "crypto_platform.h"
 /**
  * \brief   The platform context structure.
  *
  * \note    This structure may be used to assist platform-specific
  *          setup or teardown operations.
  */
 typedef struct {
-    cc_platform_ctx platform_impl_ctx; /** A context holding all the partner's platform specific context */
-    /*
-     * Add CRYS_RND_State_t rndState; when https://github.com/ARMmbed/mbedtls/issues/1200 is supported
-     * */
+    crypto_platform_ctx platform_impl_ctx; /* A context holding all the platform specific context for cryptography. Should be defined in crypto_platform.h */
+    int reference_count;
 }
 mbedtls_platform_context;
 
 
+void mbedtls_platform_init( mbedtls_platform_context* ctx);
```

Comment:
Where did this function declaration come from? Do we need it here? Where are the Doxygen comments for it? Why isn't `mbedtls_platform_teardown()` in this header as well?

In the header file that documents `mbedtls_platform_setup()` and `mbedtls_platform_teardown()`, we should comment as to their thread-safety. (As in, if it is safe to call the functions concurrently from multiple threads.)

Reply:
this is a residue from previous work, it should be removed

Reply:
>In the header file that documents mbedtls_platform_setup() and mbedtls_platform_teardown(), we should comment as to their thread-safety. (As in, if it is safe to call the functions concurrently from multiple threads.)

Probably, but this should be done as a PR in the Mbed TLS repository

Reply:
`mbedtls_platform_setup()` and `mbedtls_platform_teardown()` are part of the Mbed TLS platform.h API, and described there